django/http/response.py

```diff
@@ -408,6 +409,15 @@ def content(self, value):
             content = self.make_bytes(value)
         # Create a list of properly encoded bytestrings to support write().
         self._container = [content]
+        self._clear_cached_text()
+
+    @cached_property
+    def text(self):
+        return self.content.decode(self.charset or "utf-8")
+
+    def _clear_cached_text(self):
+        if (text := "text") in self.__dict__:
+            del self.__dict__[text]
```

Comment:
What do you think about exposing this functionality on `cached_property` instead of implementing it here for this property only?

Reply:
I intended to implement the method as an instance method within the class, but I accidentally defined it as a standalone function. Thank you for the guidance

Reply:
Are you suggesting that I implement it as an instance method within the `cached_property` class?

Reply:
Well I think we should either do that or not use cachedproperty at all. You might want to wait for a second opinion before taking this suggestion, though.

Reply:
Thank you for suggesting a good direction. I’m considering creating a related issue ticket to gather opinions from others. Would that be okay?

Reply:
Sounds wise!

Reply:
I have created an issue ticket to explore the possibility of adding a method to cached_property (see related ticket: [#35743](https://code.djangoproject.com/ticket/35743)).

However, I also received suggestions about replacing this functionality with Python 3.11's functools.cached_property. I wholeheartedly agree with the concerns raised regarding potential challenges when adopting this approach (see related ticket: [#30949](https://code.djangoproject.com/ticket/30949)).

Moreover, it's important to highlight that Django has a well-established pattern for resetting cached_property, utilizing __dict__.pop (see related pattern: [Django code search for dict.pop](https://github.com/search?q=repo%3Adjango%2Fdjango+%22__dict__.pop%22&type=code)). In this regard, I applied this pattern to initialize the cached_property accordingly.

In light of these considerations, instead of adding a new method, I have received recommendations to enhance the documentation on this topic. I plan to take action by updating the relevant documentation accordingly (see related ticket: [#35745](https://code.djangoproject.com/ticket/35745)).